usage/test-framework/src/main/java/org/apache/brooklyn/test/framework/SimpleShellCommandTestImpl.java

```diff
@@ -0,0 +1,315 @@
+        setExpectedState(this, STARTING);
+        execute();
+    }
+
+    @Override
+    public void stop() {
+        LOG.debug("{} Stopping simple command", this);
+        setUpAndRunState(false, STOPPED);
+    }
+
+    @Override
+    public void restart() {
+        LOG.debug("{} Restarting simple command", this);
+        execute();
+    }
+
+    private void setUpAndRunState(boolean up, Lifecycle status) {
+        sensors().set(SERVICE_UP, up);
+        setExpectedState(this, status);
+    }
+
+    private static class Result {
+        int exitCode;
+        String stdout;
+        String stderr;
+        public Result(final ProcessTaskWrapper<Integer> job) {
+            exitCode = job.get();
+            stdout = job.getStdout().trim();
+            stderr = job.getStderr().trim();
+        }
+        public int getExitCode() {
+            return exitCode;
+        }
+        public String getStdout() {
+            return stdout;
+        }
+        public String getStderr() {
+            return stderr;
+        }
+    }
+
+    protected void handle(Result result) {
+        LOG.debug("{}, Result is {}\nwith output [\n{}\n] and error [\n{}\n]", new Object[] {
+            this, result.getExitCode(), shorten(result.getStdout()), shorten(result.getStderr())
+        });
+        AssertionSupport support = new AssertionSupport();
+        checkAssertions(support, exitCodeAssertions(), "exit code", result.getExitCode());
+        checkAssertions(support, getConfig(ASSERT_OUT), "stdout", result.getStdout());
+        checkAssertions(support, getConfig(ASSERT_ERR), "stderr", result.getStderr());
+        support.validate();
+    }
+
+    private String shorten(String text) {
+        return Strings.maxlenWithEllipsis(text, A_LINE);
+    }
+
+    public void execute() {
+        try {
+            SshMachineLocation machineLocation =
+                Machines.findUniqueMachineLocation(resolveTarget().getLocations(), SshMachineLocation.class).get();
+            executeCommand(machineLocation);
+            setUpAndRunState(true, RUNNING);
+        } catch (Throwable t) {
+            setUpAndRunState(false, ON_FIRE);
+            throw Exceptions.propagate(t);
+        }
+    }
+
+    private void executeCommand(SshMachineLocation machineLocation) {
+
+        Result result = null;
+        String downloadUrl = getConfig(DOWNLOAD_URL);
+        String command = getConfig(COMMAND);
+
+        String downloadName = DOWNLOAD_URL.getName();
+        String commandName = COMMAND.getName();
+
+        if (!(isNonBlank(downloadUrl) ^ isNonBlank(command))) {
+            throw illegal("Must specify exactly one of", downloadName, "and", commandName);
+        }
+
+        if (isNonBlank(downloadUrl)) {
+            String scriptDir = getConfig(SCRIPT_DIR);
+            String scriptPath = calculateDestPath(downloadUrl, scriptDir);
+            result = executeDownloadedScript(machineLocation, downloadUrl, scriptPath);
```

Comment:
why copy file and run instead of the code i suggested which just runs the commands?  (i think it has built-in support to write to a file and execute the file.)